### PR TITLE
#Subscribe subscribe a user to stripe payment plan

### PR DESCRIPTION
**Pull request by Lundii**
#### What does this PR do?
- allows a user subscribe to a plan
- allows a user unsubscribe from a plan
- a route to create, get, and delete subscriptions
- a route to create, get and delete plans
- a route to create and delete customer

#### Description of Task to be completed?
Created the following routes
```
POST /api/v1/subscribe  //creates a subscription
DELETE /api/v1/subscribe/:id  //delete a subscription
GET /api/v1/subscribe  //get all subscriptions

POST /api/v1/plan  //create a plan
DELETE /api/v1/plan/:id  //delete a plan
GET /api/v1/plan  //get all plans

POST /api/v1/token  //create a token

DELETE /api/v1/customer  //delete a customer
```
#### How should this be manually tested?
- first create a token using `POST /api/v1/token  //create a token`
 ```
{
            card: {
              cardNumber: 4242424242424242,
              exp_month: 12,
              exp_year: 2020,
              cvc: '123'
            }
}
```
- then create a subscription visit
```
{
          user: {
            userId: 1,
            firstname: 'test',
            lastname: 'testlastname',
            email: 'test1@testdomain.com',
            planId: 'gold',
            token: stripeToken
          }
 }
```

#### Any background context you want to provide?
No

#### What are the relevant pivotal tracker stories?
[#167896425](https://www.pivotaltracker.com/story/show/167896425)

#### Screenshots (if appropriate)
No

### CODE REVIEW

File: server/controllers/paymentController.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  /**
+   * @static
+   * @description Get a list of users
+   * @param {object} req express request object
+   * @param {object} res express response object
+   * @returns {object} returns object response
+   * @memberof UserController
+   */
+  static async createToken(req, res) {
+    const token = await createToken(req.body.card);
```
Comment (chuxmykel) [inline server/controllers/paymentController.js:150]: Really nice job guys. I'd love to test this line locally, what exactly will I be providing from my card? Is it just the `card number` or will I also be adding the `validity date` and `CVV`?

**Review by chuxmykel — APPROVED** [review]
Great Job guys. Looks good to me.

**Review by cvjude — APPROVED** [review]
LGTM

**Review by obayomi96 — APPROVED** [review]
Good job @Lundii  and @chiboycalix on this task.,You did great work pulling this off.

File: server/controllers/paymentController.js (hunk 1 above)
Comment (Lundii) [reply server/controllers/paymentController.js:150]: The card number for testing is 4242424242424242. I will add it to the description.

**Review by encodedBicoding — APPROVED** [review]
LGTM

**Review by topseySuave — APPROVED** [review]

File: server/controllers/paymentController.js (hunk 1 above)
Comment (chuxmykel) [reply server/controllers/paymentController.js:150]: Nice one bro. You guys did a really good job with the subscription.

**Review by Hadeneekeh — COMMENTED** [review]
Awesome job!

**Pull request merged**